src/main/java/com/aws/greengrass/componentmanager/plugins/docker/DockerImageDownloader.java

```diff
@@ -28,22 +29,25 @@
 import java.time.Duration;
 import java.util.Arrays;
 import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 @SuppressWarnings({"PMD.SignatureDeclareThrowsException", "PMD.AvoidCatchingGenericException",
         "PMD.AvoidInstanceofChecksInCatchClause", "PMD.AvoidRethrowingException"})
 public class DockerImageDownloader extends ArtifactDownloader {
+    static final String DOCKER_NOT_INSTALLED_ERROR_MESSAGE = "Docker engine is not installed. Install Docker and "
+            + "retry the deployment.";
 
     private final EcrAccessor ecrAccessor;
     private final DefaultDockerClient dockerClient;
     private final MqttClient mqttClient;
 
     @Setter(AccessLevel.PACKAGE)
-    private RetryUtils.RetryConfig networkIssuesRetryConfig =
+    private RetryUtils.RetryConfig infiniteAttemptsRetryConfig =
             RetryUtils.RetryConfig.builder().initialRetryInterval(Duration.ofMinutes(1L))
```

Comment:
Had copied it from other artifact downloaders, changed the initial and max intervals so it will use exponential backoff for certain no of attempts at first. For non network issues, now there's a small initial interval but for network issues I've still left it at 1min because longer intervals makes sense for offline scenarios. Also added jitter in RetryUtils but not specific to this downloader